packages/headless/src/api/commerce/product-listings/v2/product-listing-v2-params.ts

```diff
@@ -0,0 +1,92 @@
+import {FacetValueRequest} from '../../../../features/facets/facet-set/interfaces/request';
+import {AnyFacetRequest} from '../../../../features/facets/generic/interfaces/generic-facet-request';
+import {RangeValueRequest} from '../../../../features/facets/range-facets/generic/interfaces/range-facet';
+import {SortCriterion} from '../../../../features/sort/sort';
+import {
+  HTTPContentType,
+  HttpMethods,
+  PlatformClientCallOptions,
+} from '../../../platform-client';
+
+export enum Mode {
+  Live = 'live',
+  Sample = 'Sample',
+}
+
+export interface SelectedFacets
+  extends Pick<AnyFacetRequest, 'field' | 'type' | 'facetId'> {
+  values?: (FacetValueRequest | RangeValueRequest)[];
+}
+
+export interface Context {
+  user: {
+    userId?: string;
+    email?: string;
+    userIp: string;
+    userAgent: string;
+  };
+  view: {
+    url: string;
+    labels?: Map<string, string>;
+  };
+  cart?: {
+    groupId?: string;
+    productId?: string;
+    sku?: string;
+  }[];
+}
+
+export interface ProductListingV2RequestParam {
+  platformUrl: string;
+  accessToken: string;
+  organizationId: string;
+  trackingId: string;
+  locale: string;
+  mode: Mode;
+  clientId: string;
+  selectedFacets?: SelectedFacets[];
+  selectedPage?: {
+    page?: number;
+  };
+  selectedSort?: SortCriterion;
+  context: Context;
+}
+
+/**
+ * Builds a base request for calling the Product Listing API.
+ *
+ * @param req - The Product Listing request parameters.
+ * @param method - The HTTP method used to issue the request.
+ * @param contentType - The request content type.
+ * @param queryStringArguments - The arguments to pass in the query string.
+ * @returns The built request information.
+ */
+export const baseProductListingV2Request = (
+  req: ProductListingV2RequestParam,
+  method: HttpMethods,
+  contentType: HTTPContentType,
+  queryStringArguments: Record<string, string> = {}
+): Pick<
+  PlatformClientCallOptions,
+  'accessToken' | 'method' | 'contentType' | 'url' | 'origin'
+> => {
+  const {platformUrl, organizationId, accessToken} = req;
+  const {trackingId} = req;
```

Comment:
Can we deconstruct the `trackingId` and the other params at once?
```suggestion
  const {platformUrl, organizationId, accessToken, trackingId} = req;
```